simple_parsing/parsing.py

```diff
@@ -636,17 +636,37 @@ def _resolve_subgroups(
             # Do rounds of parsing with just the subgroup arguments, until all the subgroups
             # are resolved to a dataclass type.
             logger.debug(
-                f"Starting subgroup parsing round {current_nesting_level}: {list(unresolved_subgroups.keys())}"
+                f"Starting subgroup parsing round {current_nesting_level}: "
+                f"{list(unresolved_subgroups.keys())}"
             )
             # Add all the unresolved subgroups arguments.
             for dest, subgroup_field in unresolved_subgroups.items():
                 flags = subgroup_field.option_strings
                 argument_options = subgroup_field.arg_options
 
+                # Sanity checks:
                 if subgroup_field.subgroup_default is dataclasses.MISSING:
                     assert argument_options["required"]
+                elif isinstance(argument_options["default"], dict):
+                    # TODO: In this case here, the value of a nested subgroup in this default dict
+                    # should also be used!
+                    # BUG #276: The default here is a dict because it came from a config file.
+                    # Here we want the subgroup field to have a 'str' default, because we just want
+                    # to be able to choose between the subgroup names.
+                    _default = argument_options["default"]
+                    _default_key = _infer_subgroup_key_to_use_from_config(
+                        default_in_config=_default,
+                        # subgroup_default=subgroup_field.subgroup_default,
+                        subgroup_choices=subgroup_field.subgroup_choices,
+                    )
+                    # We'd like this field to (at least temporarily) have a different default
+                    # value that is the subgroup key instead of the dictionary.
+                    argument_options["default"] = _default_key
+
                 else:
-                    assert argument_options["default"] is subgroup_field.subgroup_default
+                    assert (
+                        argument_options["default"] is subgroup_field.subgroup_default
+                    ), argument_options["default"]
```

Comment:
Hey @lebrice, thanks for your quick turn around time and hope you enjoyed NeurIPS!

For some reason, our internal tests are failing on this assertion. When I comment it out, nothing bad seems to happen and the test cases pass. This is because there are some places in our config YAMLs that we wrote:
```yml
a: KEY
```
instead of
```yml
a:
    _type_: ValueType
```
Is this intended? If so, I can update the configs on our side to work with this PR, but bringing it up in case you want to support the former way to specify subgroups that do not need any additional fields.

Reply:
Hey @lebrice, happy new year! Any updates on this?

Reply:
Yeah I'm working on it. This subgroups feature is in need of some proper refactoring. I'd be tempted to add your fixes and make a quick patch, but there are some underlying issues with the implementation that seem to warrant a broader refactor of that feature.

Reply:
Hey @lebrice, any updates on the feature refactor?

Reply:
Hey @anivegesana, thansk for the reminder. Sorry, no updates as of yet on my end. I'll let you know when I get to this (hopefully in the coming weeks).